test/mysql_tests.erl

```diff
@@ -182,8 +183,12 @@ reset_connection_test() ->
     ok = mysql:query(Pid, ?create_table_t),
     ok = mysql:query(Pid, <<"INSERT INTO t (id, tx) VALUES (1, 'text 1')">>),
     ?assertEqual(1, mysql:insert_id(Pid)),  %% auto_increment starts from 1
-    ok = mysql:reset_connection(Pid),
-    ?assertEqual(0, mysql:insert_id(Pid)), %% insertid reset to 0
+    case mysql:reset_connection(Pid) of
+      ok ->
+        ?assertEqual(0, mysql:insert_id(Pid)); %% insertid reset to 0;
+      _Error ->
+        ?assertEqual(1, mysql:insert_id(Pid)) %% reset failed
+    end,
```

Comment:
If reset_connection returns an error, how do we know it is because the server doesn't support it or because the function doesn't work as expected?

For >= 5.7.0, it should return `ok` and for < 5.7.0 maybe we don't care, or we accept both error and OK, but at we can check that it doesn't crash. To check the server version, something like `Version = db_version_string(Pid), case Version < [5, 7, 0] orelse is_mariadb(Version) of ...` can be used. Look at the JSON test case for example.

Reply:
Version check is a good idea!

Warn (or error) message is necessary (I think). Developers want to know if it is effective and what happened  when the function `reset_connection` is called.

Reply:
Version check is unavailable now.

This driver do not distinguish the difference between **MySQL** and **MariaDB**(Maybe I can try to do the work later).

- mysql server version

    ```bash
    ##myser command:status
    Server version:  5.7.27 MySQL Community Server (GPL)
    ```
    ```erlang
    %%mysql_protocol:nulterm_str/1 recved
    <53,46,55,46,50,55,0,150,0,0,0,75,110,40,2,15,112,35,52,0,255,255,8,2,0,255,193,21,0,...>>
    ```

- MariaDB server version:

    ```bash
    Server version:  5.5.5-10.4.8-MariaDB-1:10.4.8+maria~bionic mariadb.org
    ```
    ```erlang
    %%mysql_protocol:nulterm_str/1 recved
    <<"5.5.5-10.4.8-MariaDB-1:10.4.8+maria~bionic"...>>
    ```

We should use `10.4.8` in MariaDB and `5.7.27` in MySQL, both database **Type** and **Version** will be check.

So I suggest mysql-otp execute ’COM_RESET_CONNECTION‘, and check the response. Only by this way, we can get the real result.

Reply:
OK, good investigation! I'm OK with execute COM_RESET_CONNECTION and check the response for this PR. If you want to implement distinguishing between MySQL and MariaDB versions later, it would be awesome.

I would like to solve the failing deadlock test though, so the build passes on travis. I suspect the MySQL version on Travis has changed and it doesn't behave the same as before with deadlocks. I don't have time to investigate this now though.